src/style-util.js

```diff
@@ -163,23 +216,25 @@ export function processVariableAndFallback(str, callback) {
   return callback(prefix, value, fallback, suffix);
 }
 
+/**
+ * @param {Element} element
+ * @param {string} value
+ */
 export function setElementClassRaw(element, value) {
   // use native setAttribute provided by ShadyDOM when setAttribute is patched
-  if (window.ShadyDOM) {
-    window.ShadyDOM.nativeMethods.setAttribute.call(element, 'class', value);
+  if (window['ShadyDOM']) {
+    window['ShadyDOM']['nativeMethods']['setAttribute'].call(element, 'class', value);
   } else {
     element.setAttribute('class', value);
   }
 }
 
-export let rx = {
-  VAR_ASSIGN: /(?:^|[;\s{]\s*)(--[\w-]*?)\s*:\s*(?:([^;{]*)|{([^}]*)})(?:(?=[;\s}])|$)/gi,
-  MIXIN_MATCH: /(?:^|\W+)@apply\s*\(?([^);\n]*)\)?/gi,
-  VAR_CONSUMED: /(--[\w-]+)\s*([:,;)]|$)/gi,
-  ANIMATION_MATCH: /(animation\s*:)|(animation-name\s*:)/,
-  MEDIA_MATCH: /@media[^(]*(\([^)]*\))/,
-  IS_VAR: /^--/,
-  BRACKETED: /\{[^}]*\}/g,
-  HOST_PREFIX: '(?:^|[^.#[:])',
-  HOST_SUFFIX: '($|[.:[\\s>+~])'
-}
+/**
+ * @param {Element|Object} element
+ * @return {{is: string, extends: string}}
+ */
+export function getIsExtends(element) {
+  let is = element.is || (element.getAttribute && element.getAttribute('is')) || element.localName;
```

Comment:
I'll just check that localname has `-` in it. Should be faster than `getAttribute` calls in the hot path